src/vcpkg/sourceparagraph.cpp

```diff
@@ -1327,6 +1352,12 @@ namespace vcpkg
             obj.insert(el.first.to_string(), el.second);
         }
 
+        // TODO: Actually serialize vcpkg-configuration
+        if (auto configuration = scf.core_paragraph->vcpkg_configuration.get())
+        {
+            obj.insert(ManifestDeserializer::VCPKG_CONFIGURATION, *configuration);
+        }
```

Comment:
@strega-nil-ms 

I'm a bit stumped here, since we get registry information through `Configuration::registry_set()` but the `RegistryImplementation` objects are not serializable back to their manifest form. The only way I can think to serialize back is to take the `scf.core_paragraph->vcpkg_configuration` which is already a `Json::Object` and remove any nodes (probably only comments) that are not part of the registry object schema.

Reply:
https://github.com/microsoft/vcpkg-tool/pull/103 I once had a PR for serializing the registries. Don't know if that would be helpful

Reply:
@mathisloge that looks great, I'll cherry-pick your commits

Reply:
I think you would also need some fixes that weren't in the PR mentioned, since I closed it. 
https://github.com/mathisloge/vcpkg-tool/commit/5e1d40eab3196accb1f61542f17746a5226123da and https://github.com/mathisloge/vcpkg-tool/commit/5e1d40eab3196accb1f61542f17746a5226123da